lib/elastomer/client/docs.rb

```diff
@@ -261,6 +261,20 @@ def delete_by_query(query, params = nil)
         @client.delete_by_query(query, update_params(params))
       end
 
+      # Matches a document to the queries stored in the percolator for the given
+      # index. Type is required.
+      #
+      # Examples
+      #
+      #   index.register_percolator_query 1, { :query => { :match => { :author => "pea53" } } }
+      #   index.docs(type).percolate({ :doc => { :author => "pea53" } })
+      #
+      # Returns the response body as a Hash
+      def percolate(body, params = {})
+        response = client.get '/{index}/{type}/_percolate', update_params(params, :body => body, :action => 'percolator.percolate')
```

Comment:
It's possible to percolate an existing document without a body (See the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-percolate.html#_percolating_an_existing_document)) so we should support this as well. We could add an optional id element to the URI template like `/{index}/{type}{/id}/_percolate` and allow the body to be omitted.